### ISSUE TITLE
No led function on 4.19

### ISSUE THREAD

**Issue by jlores** [ewaldc/My-Book-Live#7]
The front led of Wd My Book live not run with kernel 4.19.X

**Comment by ewaldc** [comment]
Yes, this is indeed still open on 4.19.   I tried to move to standard ATA led activity for 4.19 versus a custom led driver but have not all sorted it out.   Once I get a performance patch out for 4.19, I will get the led going.

**Comment by raenye** [comment]
I'm seeing a similar (?) behavior in OpenWrt's 5.10 and 5.15 kernels. Some of the LEDs might work after boot, but eventually they won't.
Any idea when this exactly started? is it related to a particular DTS/kernel change?

**Comment by ewaldc** [comment]
It has been a long time since I looked into the led issues after the "dead" of my development MBL.  If I recall well the multicolor led HW is a not fully standard and requires a custom (static) kernel driver. The leds are also shared with uboot which monitors the loading of the kernel etc.  So it isn't the easiest of things to get right.

The issue I hit in 4.19 was with the new led registration process, the blinking process and how the 4.19 ATA drivers implemented the led trigger process.  I think most of these issues were resolved 5.x    Hence, I would guess that the OpenWRT issues are different. I think Christian used a GPIO kernel driver approach which could have other challenges, but I would need to look into it.

**Comment by raenye** [comment]
My issue might be faulty hardware, after all.
When I have the time, I'll test with my 2nd MBL.
Thanks.
